diff --git a/build.sh b/build.sh
index 949dda52..8e6681a4 100755
--- a/build.sh
+++ b/build.sh
@@ -599,7 +599,7 @@ if [ ! -d third-party ]; then
     if [ "$MOD_LUA" != "ON" ]; then
         sed -i -e "s/luajit/ /g" ./build_ols.sh
     fi
-
+    git apply  --reject --whitespace=fix ../../build_bssl.sh.patch
     ./build_ols.sh
 
 fi
@@ -656,6 +656,9 @@ cmake -DCMAKE_BUILD_TYPE=$BUILD $CMAKE_ALPINE \
     -DMOD_SECURITY=$MOD_SECURITY \
     -DMOD_LUA=$MOD_LUA \
     ..
+# Add -loqs in link.txt
+sed -i ' 1 s/.*/&-loqs/' src/CMakeFiles/openlitespeed.dir/link.txt
+
 jobs=$(nproc)
 make -j${jobs}
 cd ..
