diff --git a/script/build_bssl.sh b/script/build_bssl.sh
index 5b76948..3edb768 100755
--- a/script/build_bssl.sh
+++ b/script/build_bssl.sh
@@ -6,7 +6,11 @@ cd src
 
 
 if [ ! -d "boringssl" ]; then
-    git clone https://github.com/google/boringssl.git
+    # download boringssl and liboqs from oqs
+    echo "Downloading oqs boringssl"
+    git clone --branch $BORINGSSL_TAG https://github.com/open-quantum-safe/boringssl.git
+
+    git clone --branch $LIBOQS_TAG --single-branch --depth 1 https://github.com/open-quantum-safe/liboqs.git
 fi
 
 if [ -d "go" ]; then
@@ -17,13 +21,10 @@ echo $PATH
 fi
 
 cd boringssl
-git reset --hard
-git checkout master
-git pull
 
 # HTTP/3 v1 and ID-34 support
 #git checkout a9670a8b476470e6f874fef3554e8059683e1413
-git checkout 9fc1c33e9c21439ce5f87855a6591a9324e569fd
+# git checkout cf8d3ad3cea51cf7184307d54f465da62b7d8408
 
 rm -rf build
 
@@ -33,8 +34,13 @@ patch -p1 < ../../patches/boringssl/bssl_inttypes.patch
 #patch -p1 < ../../patches/boringssl/bssl_no_eoed.patch
 sed -i -e "s/-Werror//" CMakeLists.txt
 
-mkdir build
-cd build
+# Build liboqs both as shared and static library
+cd ../liboqs && mkdir build-static && cd build-static && cmake -G"Ninja" -DCMAKE_INSTALL_PREFIX=/opt/third-party/src/boringssl/oqs -DOQS_USE_OPENSSL=OFF .. && ninja && ninja install
+cd .. && mkdir build && cd build && cmake -G"Ninja" -DBUILD_SHARED_LIBS=ON -DCMAKE_INSTALL_PREFIX=/opt/third-party/src/boringssl/oqs -DOQS_USE_OPENSSL=OFF .. && ninja && ninja install
+
+cp /opt/third-party/src/liboqs/build/lib/liboqs.so /usr/local/lib && ldconfig
+cd ../../boringssl && mkdir build && cd build
+
 if [ "$(uname -m)" = "aarch64" ]; then
     cmake ../ -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_C_FLAGS="-fPIC" -DCMAKE_CXX_FLAGS="-fPIC" -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++
 else
@@ -50,9 +56,11 @@ make -j4
 cd ..
 
 cp crypto/libcrypto.a ../../../lib
-cp ssl/libssl.a    ../../../lib
+cp ssl/libssl.a ../../../lib
 cp decrepit/libdecrepit.a ../../../lib
+cp ../oqs/lib/liboqs* ../../../lib
 
 cd ..
+cp -r oqs/include/oqs ../../include/
 cp -r include/openssl ../../include/
 
